src/mpid/ch4/netmod/ucx/ucx_pre.h

```diff
@@ -15,6 +15,7 @@
 
 typedef struct {
     void *req;
+    ucs_status_t status;
```

Comment:
Since both field won't be in use at the same time, wouldn't a union fits better here?

Separately, I am a little bothered by the name "req". If it is more of a status info rather than a request, right?

So how about
```
typedef union {
    void * info;
    ucs_status_t status;
} MPIDI_UCX_ucp_request_t;
```

Reply:
I considered a union. I wonder if there is some remote chance that a valid request pointer could be mistaken for a status value and crash a correct program. I kept the struct for safety.

Oh, the `req` pointer is to an `MPIR_Request`. I think that type should be changed, and probably squashed with this commit.

Reply:
Upon closer look, the `req` field can be an info object or an MPIR_Request. Let me think about this a little more...

Reply:
So it is in fact:
```
typedef union {
    MPIR_Request * req;
    ucp_tag_recv_info_t * recv_info;
    ucs_status_t status;
} ...;
```
?

Reply:
Yes, that is more accurate. Again I don't know if it is technically safe lump the status in with the pointer variables, though our testing doesn't seem to have any issue.

Reply:
Yes, that is more accurate. Again I don't know if it is technically safe lump the status in with the pointer variables, though our testing doesn't seem to have any issue.